### PR TITLE
try to return resolved artifacts even exception occured

### PR DESCRIPTION
**Pull request by keshin**

### CODE REVIEW

**Comment by keshin** [comment]
the codacy issue doesn't make sense for me:
Why do I need to provide a default case for a PartialFunction like below:
```scala
    def recoverFromResolutionException: PartialFunction[Throwable, util.List[Artifact]] = {
      case e: DependencyResolutionException =>
        e.getResult.getArtifactResults.flatMap(r => Option(r.getArtifact))
    }
```

File: rtran-maven/src/main/scala/com/ebay/rtran/maven/util/MavenUtil.scala [hunk 1]
```diff
@@ -96,6 +96,12 @@ object MavenUtil {
     val dependencyFilter = new ExclusionsDependencyFilter(
       dependency.getExclusions.map(e => s"${e.getGroupId}:${e.getArtifactId}")
     )
+
+    def recoverFromResolutionException: PartialFunction[Throwable, util.List[Artifact]] = {
```
Comment (keshin) [inline rtran-maven/src/main/scala/com/ebay/rtran/maven/util/MavenUtil.scala]: the codacy issue doesn't make sense for me:
Why do I need to provide a default case for a PartialFunction like below:
```scala
    def recoverFromResolutionException: PartialFunction[Throwable, util.List[Artifact]] = {
      case e: DependencyResolutionException =>
        e.getResult.getArtifactResults.flatMap(r => Option(r.getArtifact))
    }
```

Comment (zhuchenwang) [inline rtran-maven/src/main/scala/com/ebay/rtran/maven/util/MavenUtil.scala]: Actually it does make sense. You PartialFunction is defined as [Throwable, util.List[Artifact]]. However, the case you matches is only for `DependencyResolutionException` which is only a subclass of `Throwable`. If other exceptions occurs, you will get a match exception. You can do
```scala

def recoverFromResolutionException: PartialFunction[Throwable, util.List[Artifact]] = {
      case e: DependencyResolutionException =>
        e.getResult.getArtifactResults.flatMap(r => Option(r.getArtifact))
      case other: Throwable => throw other;
    }
```

Comment (keshin) [inline rtran-maven/src/main/scala/com/ebay/rtran/maven/util/MavenUtil.scala]: Here is the detail of `recover` from `Try`, I don't think we need to declare `other` for avoiding match exception.
```scala
  def recover[U >: T](rescueException: PartialFunction[Throwable, U]): Try[U] =
    try {
      if (rescueException isDefinedAt exception) {
        Try(rescueException(exception))
      } else this
    } catch {
      case NonFatal(e) => Failure(e)
    }
```

**Pull request merged**